packages/react-dom-bindings/src/server/ReactFizzConfigDOM.js

```diff
@@ -2759,9 +2760,23 @@ function pushStyle(
         rules: ([]: Array<Chunk | PrecomputedChunk>),
         hrefs: [stringToChunk(escapeTextForBrowser(href))],
         sheets: (new Map(): Map<string, StylesheetResource>),
+        nonce: nonce && stringToChunk(escapeTextForBrowser(nonce)),
       };
       renderState.styles.set(precedence, styleQueue);
     } else {
+      if (!('nonce' in styleQueue)) {
+        // `styleQueue` could have been created by `preinit` where `nonce` is not required
+        styleQueue.nonce = nonce && stringToChunk(escapeTextForBrowser(nonce));
+      }
+      if (__DEV__) {
+        if (nonce !== styleQueue.nonce) {
```

Comment:
this won't work b/c Chunks are sometimes objects (typed arrays). But if you do what I suggested above you can use a dev only string version of the nonce to determine if this warning is necessary